src/test/java/guitests/GuiRobot.java

```diff
@@ -42,4 +44,35 @@ public void pushKeySequence(KeySequence keySequence) {
         push((KeyCodeCombination)keySequence.getKeyCombination());
         push((KeyCodeCombination)keySequence.getSecondKeyCombination());
     }
+
+    @Override
+    public FxRobot clickOn(String query, MouseButton... buttons) {
+        //Busy waiting implementation to fix issue when app window is not brought up to screen yet.
+        int count = 0;
+        while (count < 10) {
+            try {
+                return super.clickOn(query, buttons);
+            } catch (FxRobotException e) {
+                sleep(500);
+                count++;
+            }
+        }
+        return super.clickOn(query, buttons);
```

Comment:
Either don't return anything or return GuiRobot? Otherwise the chained method call will go to FxRobot even if GuiRobot overrides it.